### PR TITLE
[KernelGen][Nvidia] Add amin operator with Triton kernel

### PR DESCRIPTION
**Pull request by XDYuanzhuLee**
## Summary
Adds a Triton kernel for `amin`. Returns the minimum value of each row of the input tensor in the given dimension dim.

## Testing
- Validated against reference on device via `to_reference(inp, True)`
- Tested on: Nvidia, Tianshu, Muxi, Ascend, Hygon

## Performance
Test command: `pytest benchmark/test_amin.py --level core` (NVIDIA H20)

### amin

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|------|--------------------|-------------------|---------|
| float16 | [1048576] | 0.013376 | 0.009616 | 1.391 |
| float16 | [64, 64] | 0.006720 | 0.005856 | 1.148 |
| float16 | [4096, 4096] | 0.026880 | 0.016928 | 1.588 |
| float16 | [64, 512, 512] | 0.049472 | 0.151904 | 0.326 |
| float16 | [1024, 1024, 1024] | 1.160192 | 9.517792 | 0.122 |
| float32 | [1048576] | 0.013024 | 0.010224 | 1.274 |
| float32 | [64, 64] | 0.006944 | 0.005920 | 1.173 |
| float32 | [4096, 4096] | 0.033472 | 0.027648 | 1.211 |
| float32 | [64, 512, 512] | 0.036288 | 0.190176 | 0.191 |
| float32 | [1024, 1024, 1024] | 1.051552 | 10.009920 | 0.105 |
| bfloat16 | [1048576] | 0.013184 | 0.009696 | 1.360 |
| bfloat16 | [64, 64] | 0.006912 | 0.005984 | 1.155 |
| bfloat16 | [4096, 4096] | 0.027360 | 0.018496 | 1.479 |
| bfloat16 | [64, 512, 512] | 0.049792 | 0.159056 | 0.313 |
| bfloat16 | [1024, 1024, 1024] | 1.227040 | 9.713424 | 0.126 |

| Operator | Arithmetic Mean Speedup |
|----------|------------------------|
| amin | **0.864** |

### amin_ (in-place)

| dtype | Size | Torch Latency (ms) | Gems Latency (ms) | Speedup |
|-------|------|--------------------|-------------------|---------|
| float16 | [1048576] | 0.017376 | 0.012256 | 1.418 |
| float16 | [64, 64] | 0.009504 | 0.008384 | 1.134 |
| float16 | [4096, 4096] | 0.071840 | 0.061760 | 1.163 |
| float16 | [64, 512, 512] | 0.100960 | 0.202592 | 0.498 |
| float16 | [1024, 1024, 1024] | 4.454528 | 11.767120 | 0.379 |
| float32 | [1048576] | 0.018144 | 0.015808 | 1.148 |
| float32 | [64, 64] | 0.009216 | 0.008128 | 1.134 |
| float32 | [4096, 4096] | 0.082112 | 0.077408 | 1.061 |
| float32 | [64, 512, 512] | 0.097808 | 0.227616 | 0.430 |
| float32 | [1024, 1024, 1024] | 4.715184 | 13.135040 | 0.359 |
| bfloat16 | [1048576] | 0.017504 | 0.013120 | 1.334 |
| bfloat16 | [64, 64] | 0.009536 | 0.008384 | 1.137 |
| bfloat16 | [4096, 4096] | 0.072096 | 0.063360 | 1.138 |
| bfloat16 | [64, 512, 512] | 0.101088 | 0.210912 | 0.479 |
| bfloat16 | [1024, 1024, 1024] | 4.558720 | 11.986400 | 0.380 |

| Operator | Arithmetic Mean Speedup |
|----------|------------------------|
| amin_ (in-place) | **0.879** |

## Multi-backend Testing
| Backend | Accuracy Test | Speedup (mean) | Notes |
|---|---|---|---|
| Nvidia (H20) | PASS (30 cases) | 0.864/0.879 | Primary |
| Tianshu | PASS | 2.429 | — |
| Muxi | PASS | 0.994 | — |
| Ascend | PASS | — | — |
| Hygon | PASS | 1.114 | — |

## Files Changed
- `src/flag_gems/ops/amin.py`: Triton kernel implementation
- `tests/test_amin.py`: Accuracy test
- `benchmark/test_amin.py`: Performance benchmark
- `src/flag_gems/ops/__init__.py`: Register import and `__all__`
- `src/flag_gems/__init__.py`: Register to `_FULL_CONFIG`
- `conf/operators.yaml`: Add operator entry (kind: Reduction, stage: alpha 5.1)

### CODE REVIEW

File: tests/test_amin.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+@pytest.mark.parametrize("keepdim, dim, shape", KEEPDIM_DIMS_SHAPE)
+@pytest.mark.parametrize("dtype", utils.FLOAT_DTYPES)
+def test_amin_(shape, dim, keepdim, dtype):
+    inp = torch.randn(shape, dtype=dtype, device=flag_gems.device)
+    ref_inp = utils.to_reference(inp)
+
+    ref_out = torch.amin(ref_inp, dim=dim, keepdim=True)
+    ref_out = ref_out.expand_as(inp)
+    with flag_gems.use_gems():
+        flag_gems.amin_(inp, dim=dim, keepdim=keepdim)
```
Comment (w1120029931-bit) [inline tests/test_amin.py]: should be torch.amin_

**Review by Dongxu-H — COMMENTED** [review]
## FlagGems Operator Contribution Review

**Operator**: `amin`  
**PR**: #3909

### 📊 Change Statistics
| Metric | Value |
|--------|-------|
| Additions | 299 |
| Deletions | 0 |
| Files Changed | 6 |

### 📋 22-Point Checklist

| # | Requirement | Status |
|---|-------------|--------|
| 1 | Implementation file path correctness | ✅ Correct: `src/flag_gems/ops/amin.py` |
| 2 | KernelGen attribution in header | ✅ Present |
| 3 | Import order (stdlib → third-party → internal) | ⚠️ Could not verify |
| 4 | Function signature alignment with PyTorch API | ⚠️ Needs manual verification |
| 5 | Type support documentation | ✅ Type handling present |
| 6 | Triton kernel decorators (`@triton.jit`, etc.) | ⚠️ `@triton.jit` present, verify other decorators |
| 7 | Test file existence | ✅ Present: `tests/test_amin.py` |
| 8 | Test coverage (shapes, dtypes) | ✅ Shape and dtype tests present |
| 9 | Benchmark file existence | ✅ Present: `benchmark/test_amin.py` |
| 10 | Benchmark implementation correctness | ✅ Benchmark implemented |
| 11 | Config registration (`operators.yaml`) | ✅ Registered in `operators.yaml` |
| 12 | Module registration (`__init__.py`) | ✅ Registered in `__init__.py` |
| 13 | Ops module registration (`ops/__init__.py`) | ✅ Registered in `ops/__init__.py` |
| 14 | Logger namespace (`flag_gems.*`) | ⚠️ Missing `flag_gems.` prefix |
| 15 | Error handling | ✅ Error handling present |
| 16 | Numerical stability patterns | ⚠️ Needs domain expert review |
| 17 | Memory efficiency | ⚠️ Needs profiling |
| 18 | Edge case tests | ⚠️ Needs verification |
| 19 | Backend support (Metax) | ❌ Not implemented |
| 20 | Documentation/Docstrings | ⚠️ Needs manual review |
| 21 | Code style consistency | ⚠️ Run linter manually |
| 22 | Integration verification | ⚠️ Verify end-to-end |

### 📈 Summary

| Category | Count |
|----------|-------|
| ✅ Pass | 11 |
| ⚠️ Warning | 10 |
| ❌ Fail | 1 |

**Overall**: ⚠️ Needs Attention

---

*This review was generated by an automated script following the FlagGems Operator Contribution Guidelines.*
*For detailed code review, please consult a domain expert.*

**Review by Dongxu-H — COMMENTED** [review]
## 💡 审查建议

建议:
- ⚠️ 建议在flag_gems/__init__.py注册

**Review by Dongxu-H — COMMENTED** [review]
## 💡 审查建议

建议:
- ⚠️ 建议在flag_gems/__init__.py注册

**Review by Dongxu-H — COMMENTED** [review]
## 💡 审查建议

建议:
- ⚠️ 建议在flag_gems/__init__.py注册

**Review by XDYuanzhuLee — COMMENTED** [review]
The in-place test now uses `torch.amin` (through dispatch) with `keepdim=True` and expands the result. This matches the computation path that the `amin_` in-place implementation uses internally. (`torch.amin_` does not exist in PyTorch as a standalone function or tensor method.)

File: tests/test_amin.py (hunk 1 above)
Comment (XDYuanzhuLee) [inline tests/test_amin.py]: The in-place test now uses `torch.amin` via dispatch instead of `flag_gems.amin_` directly. Since `torch.amin_` does not exist in PyTorch, `torch.amin` with `keepdim=True` + expand tests the same computation path the in-place implementation uses internally.

Comment (XDYuanzhuLee) [reply tests/test_amin.py]: Thanks for the catch. Fixed: in-place test now uses `torch.amin` via dispatch path with `keepdim=True` + expand. Note: `torch.amin_` does not exist as a standalone PyTorch function. Fix at XDYuanzhuLee/FlagGems:pr/amin-v2 (commit 4d462e70c).

Comment (XDYuanzhuLee) [reply tests/test_amin.py]: Fix cherry-picked to correct PR branch (pr/amin). The in-place test now uses torch.amin dispatch with keepdim=True + expand.

**Review by Caeruleann — APPROVED** [review]

**Review by Dongxu-H — COMMENTED** [review]
**Bug: Assertion always passes** (line 136 in amin.py)
   `assert ((i >= -inp.ndim and i < inp.ndim) for i in dim)` uses a generator expression - generators are always truthy, so this assertion never fails for invalid dim values.
   
   **Fix**: Use `all()`:
   ```python
   assert all(i >= -inp.ndim and i < inp.ndim for i in dim), "Invalid dim"
   ```
